compiler/rustc_hir_analysis/src/lib.rs

```diff
@@ -170,29 +169,22 @@ pub fn check_crate(tcx: TyCtxt<'_>) -> Result<(), ErrorGuaranteed> {
     }
 
     tcx.sess.time("coherence_checking", || {
-        // Check impls constrain their parameters
-        let res =
-            tcx.hir().try_par_for_each_module(|module| tcx.ensure().check_mod_impl_wf(module));
+        tcx.hir().par_for_each_module(|module| {
+            let _ = tcx.ensure().check_mod_type_wf(module);
+        });
 
         for &trait_def_id in tcx.all_local_trait_impls(()).keys() {
             let _ = tcx.ensure().coherent_trait(trait_def_id);
         }
         // these queries are executed for side-effects (error reporting):
         let _ = tcx.ensure().crate_inherent_impls(());
         let _ = tcx.ensure().crate_inherent_impls_overlap_check(());
-        res
-    })?;
+    });
```

Comment:
Just removing this early abort causes annoying diagnostics that are hard to make good. So what I did instead was to inline the `check_mod_impl_wf` checks directly into the individual `check_well_formed` of impl items and early abort *per item*. So that only items with errors in `impl_wf` checks don't get their `type_wf` run, but other items do